src/network/mod.rs

```diff
@@ -309,17 +337,17 @@ fn first_trusted_peer_address(config: &Configuration) -> Option<SocketAddr> {
         .next()
 }
 
-pub fn bootstrap(config: &Configuration, blockchain: BlockchainR) {
+pub fn bootstrap(config: &Configuration, blockchain: BlockchainR, logger: &Logger) {
     if config.protocol != Protocol::Grpc {
         unimplemented!()
     }
     match first_trusted_peer_address(config) {
         Some(address) => {
             let peer = Peer::new(address, Protocol::Grpc);
-            grpc::bootstrap_from_peer(peer, blockchain)
+            grpc::bootstrap_from_peer(peer, blockchain, logger)
```

Comment:
Make a clone with the peer address and a "bootstrap" task tag?

Reply:
It's already done in `main::bootstrap`, line 224. Thanks to that all actions in bootstrapping phase have this tag. It's insanely hard to follow, we have to think about some way to make it clearer.

Reply:
Right, thanks.